### ISSUE TITLE
Testing of Hardware

### ISSUE THREAD

**Issue by zrajani** [zrajani/iLETTERBOX_CAPSTONE_PROJECT_EMBT_F2020#6]
For the components that were received before one can use them one needs to do proper testing to ensure that it working properly and provides the desired voltage and current as required by the project

**Comment by zrajani** [comment]
The testing started with a small bug: Arduino was unable to recognize (read) the USB attached for the Arduino in the Linux operating system. The solution is being searched on the Internet

**Comment by zrajani** [comment]
The following command resolved the issue: sudo chmod a+rw /dev/ttyACM0

**Comment by zrajani** [comment]
The following Issues also occurred during the testing phase:
1. GSM Module connection: Initially the GSM Module wasn't able to connect but giving some time and using the power key it started working 
2. BB-AI after getting connected to WiFi and during the update went into a sudden shutdown. This was suddenly resolved after pressing the reset button. Also, the processor started getting hot but later cooled down itself.

**Comment by zrajani** [comment]
The codes for testing the components were added successfully to the repository. SHortly will add the various commands used to set up the WiFi connection in the Beaglebone AI hardware

**Comment by mikeshams** [comment]
> The following Issues also occurred during the testing phase:
> 
> 1. GSM Module connection: Initially the GSM Module wasn't able to connect but giving some time and using the power key it started worki

What do you mean by the power key?

**Comment by zrajani** [comment]
There is a key or a button next to the power LED on the shield. This key helps me to connect the Module to the desired network. Like when I give a long press on this button the Status and Network LED glow continuously and when connected to the network, then network LED blinks every 3 seconds and Status LED remains stable i.e. ON always

This link helps one understand better on how to start with the module and use power key: https://randomnerdtutorials.com/sim900-gsm-gprs-shield-arduino/

**Comment by zrajani** [comment]
> The following Issues also occurred during the testing phase:
> 
> 1. GSM Module connection: Initially the GSM Module wasn't able to connect but giving some time and using the power key it started working
> 2. BB-AI after getting connected to WiFi and during the update went into a sudden shutdown. This was suddenly resolved after pressing the reset button. Also, the processor started getting hot but later cooled down itself.

Since the BB-AI is getting hot every time we have ordered a BB-AI Fan cape from Newark Canada I hoping it to get it delivered by end of the month. Here's is a link if one wishes to buy: [BB-AI Fan Cape](https://canada.newark.com/element14/6100310/beaglebone-ai-fan-cape/dp/50AH3704)

**Comment by zrajani** [comment]
As the heating of the BB-AI was going out of control and cannot be cooled with the original fan cape  of BB and the shutdown became too frequent thus after speaking to the technical team at ELEMENT14 (manufacturer) decided to move on with Beaglebone Wireless. Approval taken already!

**Comment by zrajani** [comment]
> This shall be posted in the task issue Interfacing IR with B as well. 

The Beaglebone Wireless Datasheet has been attached and now the communication with the BeagleBone shall occur at IP address 198.162.8.1 as the Module has in-built Wifi but can also at the regular IP but the IP mentioned is for wireless communication with the board. For this purpose also the host machine through which we intend to login in shall be connected to the Beaglebone Wifi. After login, we are expected to go into the Cloud9 IDE where one can easily program and work with the terminal and the desktop of the BB-WI. The hardware is expected to arrive by Friday evening. Testing of the hardware is expected to occur on the same day as the next Task will be scheduled in the following week.

**Comment by zrajani** [comment]
>This shall be posted in the task issue Interfacing IR with BB-WI as well.

The Beaglebone Wireless Datasheet has been attached and now the communication with the BeagleBone shall occur at IP address 198.162.8.1 as the Module has in-built Wifi but can also at the regular IP but the IP mentioned is for wireless communication with the board. For this purpose also the host machine through which we intend to login in shall be connected to the Beaglebone Wifi. After login, we are expected to go into the Cloud9 IDE where one can easily program and work with the terminal and the desktop of the BB-WI. The hardware is expected to arrive by Friday evening. Testing of the hardware is expected to occur on the same day as the next Task will be scheduled in the following week.

**Comment by zrajani** [comment]
> As the heating of the BB-AI was going out of control and cannot be cooled with the original fan cape of BB and the shutdown became too frequent thus after speaking to the technical team at ELEMENT14 (manufacturer) decided to move on with Beaglebone Wireless. Approval taken already!

Error rectified we did a mistake of uploading a wrong  image file on the BB-AI as the image that we uploaded was only for BB-WI, x15, BBB, etc. Thus because of wrong flashing of the image the BB-AI was not able to connect to the WIFI and perform the task properly. Thus when proper image was flashed the BB-AI was able to perform properly. As for the heating issue it continued to heat up but was in control but since we had a fan cape with us which was for the BB-AI specifically when attached with the same the heat was also resolved the BB-AI now stays at normal temperature and doesn't get hotter any longer. Thus, these issues were finally resolved but since its too late now to make a change hence the project continues with BB-WI and we shall use the BB-AI in some of our future projects
